turtle.js

```diff
@@ -0,0 +1,138 @@
+          for (let i = 0; i < steps; i++) {
+          this.y += 1;   
+          this.coordinates.push([this.x, this.y])
+          }
+     } else if (this.face === 'west') {
+          for (let i = 0; i < steps; i++) {
+          this.x += -1;
+          this.coordinates.push([this.x, this.y])
+          }
+     } else if (this.face === 'north') {
+          for (let i = 0; i < steps; i++) {
+          this.y += -1
+          this.coordinates.push([this.x, this.y])
+          }
+     }
+   }
+   right(){
+        if (this.face === 'east') {
+             this.face = 'south'
+        } else if (this.face === 'south') {
+             this.face = 'west'
+        } else if (this.face === 'west') {
+             this.face = 'north'
+        } else if (this.face === 'north') {
+             this.face = 'east'
+        }
+     }
+
+   left(){
+     if (this.face === 'east') {
+          this.face = 'north'
+     } else if (this.face === 'north') {
+          this.face = 'west'
+     } else if (this.face === 'west') {
+          this.face = 'south'
+     } else if (this.face === 'south') {
+          this.face = 'east'
+     }
+   }  
+
+   print(){
+     let allCoordinates = this.coordinates;
+     let maxCoordinates = [];
+     let board = [];
+     let boardString = '';
+ 
+     function findMax (x){
+       let biggestX = 0
+       let biggestY = 0
+       for (let i = 0; i < x.length; i++){
+          if(x[i][0] > biggestX){
+            biggestX = x[i][0]
+          }
+         }
+       for (let j = 0; j < x.length; j++){
+          if(x[j][1] > biggestY){
+            biggestY = x[j][1]
+          }
+         }
+      
+        maxCoordinates.push([biggestX, biggestY])
+        
+     }
+      findMax(allCoordinates)
+      // return maxCoordinates
+      
+     
+     function makeBoard(width, height){
+       for (let i = 0; i<=height; i++){
+         for (let j = 0; j<=width; j++){
+          board.push([j, i])
+         }
+       }
+     }
+      makeBoard(maxCoordinates[0][0], maxCoordinates[0][1])
+      // return board
+      function printBoard(board, width){
+          // console.log(board.length)
+          for(let boardPoint = 0; boardPoint < board.length; boardPoint++ ){
+          let currentPoint = board[boardPoint];
+          let isTurtleStep = allCoordinates.find((step) => {
+              // console.log(step[0], 'step')
+              // console.log(currentPoint[0], 'currentPoint')
+              return (step[0] === currentPoint[0]) && (step[1] === currentPoint[1])
+            })
+            if (currentPoint[0] === width){  // width - 1?
+              boardString += isTurtleStep ? '* \n' : '0 \n';
+            } else {
+              boardString += isTurtleStep ? '*' : '0';
+            }
+          } 
+          // console.log(boardString)
+        }
+        printBoard(board, maxCoordinates[0][0]) 
+        console.log(boardString) 
+     }
+  
+}
+
+let Sonic = new Turtle (0,0)
+Sonic
+Sonic.forward(5)
+Sonic.right()
```

Comment:
Very good work around. I know you were unable to get the following to work
```js
new Turtle(0, 0)
  .forward(5)
  .right()
  .forward(5)
  .right()
  .forward(5)
  .right()
  .forward(5)
  .print()
```
using chaining. 

The reason this wasn't working was because you could only call `right` or `forward` (or any other method belonging to the turtle class) on an instance of a turtle.
I can see in your work around, that's exactly what you did. You realized that chaining wasn't working so you used your turtle instance (`Sonic`) to call the methods one at a time.

In order to get chaining to work, you need to make each of those methods described above (`right`, `left`, `forward`) have a return value.
Currently, since there is no explicit return statement, each method is implicitly returning `undefined`.
Because of this implicit return of `undefined`, when you chain a `right` onto a call to `forward` like below:
```js
new Turtle (0,0)
    .forward(5)
    .right()
```
JS/Node thinks you are trying to invoke the `right` method on this implicitly returned `undefined`.
It can be thought of like this:
```js
const turtle = new Turtle(0,0);
const turtle1 = turtle.forward(5); // turtle1 becomes undefined
turtle1.right()
```
 
So, what we would like is for each of those methods to return the turtle.
To do this, we need each method to return `this` at the end of it.
Like this:
```js
 forward(steps) {
    // ...
     return this;
  }

 right() {
    // ...
     return this;
  }

 left() {
    // ...
     return this;
  }
```
And now chaining will work!

Reply:
Thanks Max.  Changes made. I can't believe i didn't think about that!